### DISCUSSION TITLE
Enterprise Fil+ WG - Overview

### DISCUSSION THREAD

**Discussion by dkkapur** [filecoin-project/notary-governance#527] [E-Fil+ WG]
@megtei (Notary from Holon) is leading an investigating into the path to DataCap for enterprise use cases. 

See few minutes of this clip: https://youtu.be/6z5kk6pPwhg?t=1996. 

Specifically, this means in cases that we want to serve clients that do not fit into the usual "open data" or "open KYC", i.e., where the data is encrypted or the client name cannot be revealed publicly, we need a better path for those clients to prove their trustiworthiness and receive DataCap.  This WG effort is within the Fil+ community and system, and aims to track the path to DataCap, identify key pain points for enterprise clients, and recommend changes to how we (community, notaries) support enterprise clients applications for DataCap.

**Comment by dkkapur** [comment]
tracking updates on the client KYC front here: https://github.com/filecoin-project/notary-governance/discussions/528

**Comment by MegTei** [comment]
[Fil+ day_Enterprise Program_proposalV0.1 .pdf](https://github.com/filecoin-project/notary-governance/files/8819412/Fil%2B.day_Enterprise.Program_proposalV0.1.pdf)
Hi everyone, find the short overview of the enterprise proposal. This will be presented 7-Jun at Fil+ day.
Consensus or feedback is invited.
If you are providing feedback, please be sure its constructive, include alternatives/ ideas (no venting without action!)

**Reply by Tom-OriginStorage** [reply]
One issue that I am worried about which has been mentioned in the [Enterprise Program Governance ProposalV0.1.pdf](https://github.com/filecoin-project/notary-governance/files/8819416/Enterprise.Program.Governance.ProposalV0.1.pdf) is the lack of incentives for clients to continue the Enterprise Program after receiving their complete share of Datacap.

Clients may faced abrupt choices after the end of the Enterprise Program:
1. Apply for the Enterprise Program again to continue their free storage
2. Give up on Enterprise Program and return back to old alternatives
3. Start using Filecoin for their storage without the boosted incentives

Ideally, we want scenario 3 to be the outcome of the Enterprise Program. As such, I think it would be great for Notaries to advice the clients and relevant SPs of Enterprise Program to slowly transit into scenario 3 by having a growing portion of the deals as unverified deals. 

For example：
1st week: 0% unverified 100% verified
2nd week: 1% unverified 99% verified
3rd week: 2% unverified 98% verified
...
Nth week: 100% unverified 0% verified (And from here on officially "graduated" from the Enterprise Program)

This would also help the client realistically estimate the costs they will actually incur when they finish the Enterprise Program and bring forth better adoption.

**Reply by kernelogic** [reply]
If the negative pricing FIP was approved, a lot of the challenges can be avoided ...

**Reply by MegTei** [reply]
HI @Tom-OriginStorage yes, managing expectations upfront is very important. Should this get approved, I'll loop you into the value proposition design

**Reply by MegTei** [reply]
HI @kernelogic Fei Yan, can you expand on negative pricing FIP and what it will achieve/ how it could contribute to an enterprise program?

**Comment by MegTei** [comment]
If you are from the governance community, the detailed proposal is here
[Enterprise Program Governance ProposalV0.1.pdf](https://github.com/filecoin-project/notary-governance/files/8819416/Enterprise.Program.Governance.ProposalV0.1.pdf)

**Comment by MegTei** [comment]
If you are from the SP community, the detailed proposal is here
[Fil-E SP Program ProposalV0.1.pdf](https://github.com/filecoin-project/notary-governance/files/8819418/Fil-E.SP.Program.ProposalV0.1.pdf)

**Comment by cryptowhizzard** [comment]
Hi @MegTei 

Let me first start with compliments. These documents look great and are very comprehensive.

For the enterprise parts i have a question:

Clients are company’s here as i understand correctly reading through your proposal? Most clients / their company’s have non IPFS specialised ICT managers and in this enterprise programs we do not want to bother them with a 12 month learning curve to be able to make or compile large datasets? Who is going to pack their data and distribute it to SP’s ? Should they be called data managers or data distributors?  Can they be incentivised by asking money to clients and/or sp’s for their work / labour done and hardware / internet costs involved getting everything on the right place?

For the KYC i think you are heading in the right direction by having a CEO/MD signing off on the storage request / contract. Combined with a COC registration check this would be a great way to go.

Having said this i am really exited reading this as this is a great basis to start with. Good job!

**Reply by MegTei** [reply]
HI @cryptowhizzard Wijnand, great questions about the data egress/ ingress stage - a conversation I'd like to loop in with Joao as he has done more analysis on the SP enablement side. Should this get approved, we'll have a session about effort/ incentives.

**Comment by davidthoms** [comment]
Hi @MegTei ,

Dave  & Job here from Coffee Cloud. We just wanted to say a very well done in your presentation yesterday. You mentioned about requiring support for Governance and Storage providers, in particular in Asia/China. We are based in Hong Kong and would be very interested in supporting this project in any way we can. I think we may work best in supporting the solution in a governance role. Any further details would be very much appreciated. We are very excited about this project and can see its immediate application within our ecosystem. Very exciting times ahead and great job!

**Comment by MRJAVAZHAO** [comment]
Hello, @MegTei
So happy to see that FIL+ large dataset has expanded into the filed of private data. We are willing to join this program and to be an active notary. In the past, we are active in verify clients. We check slack and github at least 5 days a week. We believe we can handle the task of notaries.
We think where are three core factors, client, data, storage providers. Of course, the development and operation work for the project is equally important.

1. Clients

We think we should concern more with clients themselves, whether it is a organization with real storage requirements. So we can use some automatic KYC verification procedules. It looks like deep had came up with some good ideas. That's awesome.

2. Data

As @Tom-OriginStorage said, having a growing portion of the deals as unverified deals might be a good way. But some enterprises do not have public data. So I think we should concern more with the data validity and security.  Storage providers in different regions should be concerned about this point, so can we embed the data validity check step in data uploading process.

3. SPs

The point of concern is that how to avoid storage providers clients do self-dealing. I think maybe we can create robot to automatic list about storage providers distribution, as well as the corresponding proportion, IP, area and publish these informations on the issue, and then notary can determine whether to approve the next allocation.

**Comment by galen-mcandrew** [comment]
@MegTei @mr-spaghetti-code 

So excited for the progress this working group has made! The presentations at Fil+ Day and FIL Austin were amazing. I wanted to follow up here in writing to recap some of the ad-hoc conversations that have happened, to try and align on next steps as I understand them.

High level though, I think the working group is on-track to pitch a **scoped proof of concept for enterprise data**, and we should be testing something before October!

1. WG will post a summarized implementation plan in Notary Governance Repo
    * This could come from any member of the working group or some new WG GH account
    * It should match the “[Modification](https://github.com/filecoin-project/notary-governance/issues/new?assignees=&labels=&template=modification.md&title=Modification%3A+%5BLocation+of+Change%5D+-+%5BDescription+of+Change%5D)” template (not sure that link will work, this template lives in the repo when you create a new issue)
    * Ideally a community member could read this in 5 minutes and have an understanding of the **Why, Who, How, How much, and For How Long**, as well as the ‘Things to Watch Out For’ 
    * Some examples to learn from:
        * [LDN v2](https://github.com/filecoin-project/notary-governance/issues/217)
        * [Proposal to continue the entire LDN pathway beyond initial PoC](https://github.com/filecoin-project/notary-governance/issues/328)
        *  [LDN allocation calculation changes](https://github.com/filecoin-project/notary-governance/issues/345)
        *  [Seal storage first PoC for encrypted NDA private client](https://github.com/filecoin-project/notary-governance/issues/489)
1. Gov team can weigh in on this proposal and add support/feedback, along with community
1. WG can pitch this scoped implementation plan for the proof of concept at next governance call (June 28) with a targeted roll-out plan and dates
1. If approved, WG and Gov team will be responsible for supporting implementation
    * These roles/responsibilities should be scoped in the proposal. What will be required from community in order to implement? 
    * For example, do new multisigs need to be created? Do new tools or pathways have to be created and tested? Does a new dashboard for tracking need to be implemented?

**Summary:** draft a PoC one-pager that matches the governance proposal format and can be posted after review, which is highly tactical with implementation details and end-target, get community support, run targeted PoC and evaluate learnings.

Hopefully this action plan is useful. I know navigating the open-source community-governed project space is ambiguous, and appreciate this working group generating a proposal for the program! I look forward to reading your proof of concept for how the community can safely perform diligence at the enterprise scale.

**Comment by MatrixStorage** [comment]
Hi  @MegTei 
This is a very good program that MatrixStorage would like to actively participate in.
In addition, we will keep communicating and hope to share with you some problems and experiences we met in the past in any form.

**Reply by PluskitOfficial** [reply]
Hi, @MegTei hi, community members,

This is Pluskit. As a new notary, we're very grateful to Megtei for your contribution to the community; as a Fil+ ecosystem data tool platform, we'd like to participate in the Fil-E project.

We will continue to pay attention to the progress of the project. For the current stage, our thoughts are as follows.

1. The verification of the enterprise itself is higher than the random inspection of the data itself.
The large-scale nature of enterprise-level data determines that the data itself cannot be verified in excessive detail and continuously. At the same time, the enterprise itself is a subject that pays great attention to reputation. Therefore, it is more important and feasible to verify the enterprise itself. In the early stage of the project, a small number of clients with high comprehensive scores in terms of data volume, high reputation, and establishment time etc. can be selected from a batch of applicants, rather than screening by an absolute measure.

2. We agree with the practice of contract signing
Just like in the commercial society, cooperation between enterprises needs to be constrained by very standardized legal documents, so as to ensure the implementation of the entire link and the safety of the project.

3. Refer to the current Fil+ way to avoid private transactions
Information disclosure and review such as storage nodes, storage ratios, and SPs' location could be conducted in subsequent stages to reduce the occurrence of private transactions as much as possible. **Pluskit has a lot of experience in related data recording and presentation. If necessary, we can add more data and even API for the project at any time.**

4. Using data security tools
We recommend using some automatic retrieval tools for sensitive data, including filtering, prompting and blocking of sensitive content such as sensitive words and sensitive pictures. This will effectively balances privacy and security, helping the project move forward more smoothly.

**Comment by MegTei** [comment]
Hi @all thanks for the structure @galen-mcandrew Modification has been raised [#559](https://github.com/filecoin-project/notary-governance/issues/559) 
The pilot only commences once the planning phase has been worked through, which I think is what your referring for Oct?
There is no working group as yet, expressions of interest and following feedback cycle can form both SP & Governance groups.
